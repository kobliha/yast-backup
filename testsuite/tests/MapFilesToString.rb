# encoding: utf-8

# Backup module - testsuite
#
# Backup::MapFilesToString function tests
#
# testedfiles: Backup.ycp
#
# $Id$
#
module Yast
  class MapFilesToStringClient < Client
    def main
      Yast.import "Testsuite"

      @readmap = {
        "target" => { "string" => "", "size" => -1, "tmpdir" => "/tmp" },
        "proc"   => { "cpuinfo" => { "value" => { "0" => { "flags" => "" } } } },
        "probe"  => {
          "has_smp" => false,
          "is_uml"  => false,
          "cdrom"   => { "manual" => [] }
        }
      }

      @execmap = {
        "target" => {
          "bash_output" => {
            "exit"   => 0,
            "stderr" => "",
            "stdout" => "charmap=\"ISO-8859-1\"\n"
          }
        }
      }

      Testsuite.Init([@readmap, {}, @execmap], nil)

      Yast.import "Pkg" # override packamanager
      Yast.import "Backup"

      # initialize variables from Backup module
      Backup.backup_files = nil
      Backup.selected_files = {
        "aaa_base-2001.5.22-0" =>
          #"install_prefixes" : "(none)"
          { "changed_files" => ["/etc/modules.conf"] }
      }

      Testsuite.Dump([Backup.MapFilesToString])

      Backup.backup_files = {}
      Testsuite.Dump([Backup.MapFilesToString])

      Backup.backup_files = {
        "pkg1"                 => {
          "changed_files"    => [],
          "install_prefixes" => []
        },
        "pkg2"                 => {
          "changed_files"    => nil,
          "install_prefixes" => nil
        },
        "aaa_base-2001.5.22-0" => {
          "changed_files" => ["/etc/inittab", "/etc/modules.conf"]
        },
        ""                     => {
          "changed_files" => ["/etc/fstab", "/etc/passwd"]
        },
        "pkg3"                 => nil
      }

      Backup.selected_files = Builtins.add(
        Backup.selected_files,
        "",
        { "changed_files" => ["/etc/fstab"] }
      )

      Testsuite.Dump([Backup.MapFilesToString])

      nil
    end
  end
end

Yast::MapFilesToStringClient.new.main
