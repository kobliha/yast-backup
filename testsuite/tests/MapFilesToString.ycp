// Backup module - testsuite
//
// Backup::MapFilesToString function tests
//
// $Id$
//

{

include "backup/functions.ycp";
include "testsuite.ycp";

// proc.cpuinfo.value.0.flags, target.string - used in Kernel::
// probe.has_smp - Arch::
// probe.cdrom.manual - StorageDevices::
// target.size - AutoInstall::

TESTSUITE_INIT([ $[ "target" : $["string" : "", "size" : -1, "tmpdir" : "/tmp" ],
    "proc" : $["cpuinfo" : $[ "value" : $["0" : $[ "flags" : ""]]]],
    "probe" : $["has_smp" : false, "cdrom" : $["manual" : []]]  ], $[], $[] ] ,nil);
import "Pkg";	// override packamanager
import "Backup";

// initialize variables from Backup module
Backup::backup_files = nil;
Backup::selected_files = $["aaa_base-2001.5.22-0" : $["changed_files" : ["/etc/modules.conf"], /*"install_prefixes" : "(none)"*/]];

DUMP( [ Backup::MapFilesToString() ] );

Backup::backup_files = $[];
DUMP( [ Backup::MapFilesToString() ] );

Backup::backup_files = $[
  "pkg1": $["changed_files" : [], "install_prefixes" : []],
  "pkg2" : $["changed_files" : nil, "install_prefixes" : nil],
  "aaa_base-2001.5.22-0" : $["changed_files" : ["/etc/inittab", "/etc/modules.conf"]],
  "" : $["changed_files" : ["/etc/fstab", "/etc/passwd"]],
  "pkg3" : nil
];

Backup::selected_files = add(Backup::selected_files, "", $["changed_files" : ["/etc/fstab"]]);

DUMP( [ Backup::MapFilesToString() ] );

}

