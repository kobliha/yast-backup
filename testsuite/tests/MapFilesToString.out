Read	.target.tmpdir "/tmp"
Dump	[""]
Dump	[$["filelist":"Package: aaa_base-2001.5.22-0\nInstalled: (none)\n/etc/modules.conf\n", "sel_files":1, "sel_packages":1]]
Dump	[$["filelist":"Package: aaa_base-2001.5.22-0\nInstalled: (none)\n/etc/modules.conf\nNopackage:\n/etc/fstab\n", "sel_files":2, "sel_packages":2]]
